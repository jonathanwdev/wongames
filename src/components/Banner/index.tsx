import Button from 'components/Button'
import Ribbon, { RibbonColors, RibbonSizes } from 'components/Ribbon'
import * as S from './styles'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  ribbon?: React.ReactNode
  ribbonSize?: RibbonSizes
  ribbobColor?: RibbonColors
}

const Banner = ({
  buttonLabel,
  buttonLink,
  img,
  subtitle,
  title,
  ribbon,
  ribbonSize = 'normal',
  ribbobColor = 'primary'
}: BannerProps) => (
  <S.Wrapper>
    {!!ribbon && (
      <Ribbon size={ribbonSize} color={ribbobColor}>
        {ribbon}
      </Ribbon>
    )}
    <S.Image src={img} role="img" aria-label={title} />
    <S.Caption>
      <S.Title>{title}</S.Title>
      <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
      <Button as="a" href={buttonLink} size="large">
        {buttonLabel}
      </Button>
    </S.Caption>
  </S.Wrapper>
)

export default Banner
