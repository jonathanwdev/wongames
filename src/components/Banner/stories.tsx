import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img:
      'https://s.aficionados.com.br/imagens/hunter-x-hunter-conheca-os-personagens-do-iconico-anime_f.jpg',
    title: 'Hunter x Hunter',
    subtitle:
      '<p>Watch the new <strong>Game</strong> of the perfect anime hunter x hunter</p>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/hunter-x-hunter'
  },
  argTypes: {
    ribbon: { type: 'string' }
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: Story<BannerProps> = (args) => <Banner {...args} />

export const WithRibbon: Story<BannerProps> = (args) => <Banner {...args} />

WithRibbon.args = {
  ribbon: '20% off',
  ribbonSize: 'normal',
  ribbobColor: 'primary'
}
