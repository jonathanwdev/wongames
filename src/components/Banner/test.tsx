import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'
import Ribbon from 'components/Ribbon'

const props = {
  img:
    'https://s.aficionados.com.br/imagens/hunter-x-hunter-conheca-os-personagens-do-iconico-anime_f.jpg',
  title: 'Hunter x Hunter',
  subtitle:
    '<p>Watch the new <strong>Game</strong> of the perfect anime hunter x hunter</p>',
  buttonLabel: 'Buy now',
  buttonLink: '/games/hunter-x-hunter'
}

describe('<Banner />', () => {
  it('should render correctely', () => {
    const { container } = renderWithTheme(<Banner {...props} />)
    const heading = screen.getAllByRole('heading')

    expect(heading[0]).toBeInTheDocument()
    expect(heading[1]).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /Hunter x Hunter/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render ribbon', () => {
    renderWithTheme(
      <Banner
        {...props}
        ribbon="My Ribbon"
        ribbonSize="small"
        ribbobColor="secondary"
      />
    )
    const ribbon = screen.getByText(/My Ribbon/i)
    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({
      background: '#3CD3C1'
    })
    expect(ribbon).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    })
  })
})
