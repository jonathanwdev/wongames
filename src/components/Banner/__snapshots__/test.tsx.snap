// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctely 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.8rem;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35568 0%,#d958a6 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://s.aficionados.com.br/imagens/hunter-x-hunter-conheca-os-personagens-do-iconico-anime_f.jpg);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
  margin-bottom: 0.8rem;
}

.c4 strong {
  color: #F231A5;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (max-width:1170px) {
  .c0 .sc-gsTCUz {
    right: 0;
  }

  .c0 .sc-gsTCUz::before {
    display: none;
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Hunter x Hunter"
    class="c1"
    role="img"
    src="https://s.aficionados.com.br/imagens/hunter-x-hunter-conheca-os-personagens-do-iconico-anime_f.jpg"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Hunter x Hunter
    </h2>
    <h3
      class="c4"
    >
      <p>
        Watch the new 
        <strong>
          Game
        </strong>
         of the perfect anime hunter x hunter
      </p>
    </h3>
    <a
      class="c5"
      href="/games/hunter-x-hunter"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
